pytools/graph.py

```diff
@@ -299,8 +299,8 @@ def compute_topological_order(graph: Mapping[T, Collection[T]],
 
     if len(order) != total_num_nodes:
         # any node which has a predecessor left is a part of a cycle
-        raise CycleError(next(iter(n for n, num_preds in
-            nodes_to_num_predecessors.items() if num_preds != 0)))
+        raise CycleError(list(n for n, num_preds in
+            nodes_to_num_predecessors.items() if num_preds != 0))
```

Comment:
It would be nice to provide this in traversable (rather than arbitrary) order, which is doable at $O(n)$ cost.

Reply:
I'm not sure how to do that - how would we provide this in case there are multiple cycles in the graph?

Reply:
#167 has some work in this direction